### PR TITLE
[CAKE 704] - Fixed Impact Foot Recirculation Experment discussions avatar bug

### PR DESCRIPTION
**Pull request by tiffperumpail**
@Wikia/cake 
https://wikia-inc.atlassian.net/browse/CAKE-704

Modified CakeRelatedContentService.php.getAuthor($content) to return an array including both the author's username and avatar, instead of only the author's username. Updated CakeRelatedContentHelper.formatData(data) to use new author object.

### CODE REVIEW

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js [hunk 1]
```diff
@@ -42,10 +42,14 @@ define('ext.wikia.recirculation.helpers.cakeRelatedContent', [
...
-            return item;
+			if (item.source  == "discussions") {
+				item.meta.authorAvatarUrl = item.author.avatar.src;
+			}
+			item.author = item.author.username;
+			item.title = item.title.split(' | ')[0];
+			return item;
         }).sort(utils.sortThumbnails);
-
+		debugger
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: CRITICAL Remove this debugger statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ADebuggerStatement)

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 2]
```diff
@@ -163,7 +163,8 @@ private function getAuthor( Content $content ) {
 			return "";
 		}
 
-		return $authorObjects[0]->getUsername();
+		return array("username" => $authorObjects[0]->getUsername(),
+					"avatar" => $authorObjects[0]->getAvatar());
 	}
 
 	private function getRecirculationContentType( $contentType ) {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* CRITICAL 1 critical
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/a358bc1959ab964f72f4b52d3bd5cad43273f4bb/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/a358bc1959ab964f72f4b52d3bd5cad43273f4bb/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/a358bc1959ab964f72f4b52d3bd5cad43273f4bb/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js [hunk 3]
```diff
@@ -42,10 +42,14 @@ define('ext.wikia.recirculation.helpers.cakeRelatedContent', [
 
     function formatData(data) {
         data.items = data.items.map(function(item) {
-            item.title = item.title.split(' | ')[0];
-            return item;
+			if (item.source  == "discussions") {
```
Comment (nmonterroso) [inline extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: Can you fix the spacing here?

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 4]
```diff
@@ -163,7 +163,8 @@ private function getAuthor( Content $content ) {
 			return "";
 		}
 
-		return $authorObjects[0]->getUsername();
+		return array("username" => $authorObjects[0]->getUsername(),
+					"avatar" => $authorObjects[0]->getAvatar());
```
Comment (nmonterroso) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php]: same comment here about spacing

**Review by nmonterroso — COMMENTED** [review]
some comments about spacing otw lgtm

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js (hunk 1 above)
Comment (tiffperumpail) [inline extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: Fixed in f01bc1ef0e823d2b0279f03cb97feb28bf2c356e

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/f01bc1ef0e823d2b0279f03cb97feb28bf2c356e/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/f01bc1ef0e823d2b0279f03cb97feb28bf2c356e/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/f01bc1ef0e823d2b0279f03cb97feb28bf2c356e/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js [hunk 5]
```diff
@@ -42,10 +42,13 @@ define('ext.wikia.recirculation.helpers.cakeRelatedContent', [
 
     function formatData(data) {
         data.items = data.items.map(function(item) {
-            item.title = item.title.split(' | ')[0];
-            return item;
+		if (item.source  == "discussions") {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: MAJOR Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/cf5e20b84e531121a27a4298c8ddbff5acd8ab4b/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/cf5e20b84e531121a27a4298c8ddbff5acd8ab4b/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/cf5e20b84e531121a27a4298c8ddbff5acd8ab4b/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js [hunk 6]
```diff
@@ -42,10 +42,13 @@ define('ext.wikia.recirculation.helpers.cakeRelatedContent', [
 
     function formatData(data) {
         data.items = data.items.map(function(item) {
-            item.title = item.title.split(' | ')[0];
-            return item;
+	    if (item.source  == "discussions") {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: MAJOR Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/2f79991f0654b40277978274b077c2b5d2f8f983/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/2f79991f0654b40277978274b077c2b5d2f8f983/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/2f79991f0654b40277978274b077c2b5d2f8f983/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php [hunk 7]
```diff
@@ -163,7 +163,8 @@ private function getAuthor( Content $content ) {
 			return "";
 		}
 
-		return $authorObjects[0]->getUsername();
+		return array("username" => $authorObjects[0]->getUsername(),
+		             "avatar" => $authorObjects[0]->getAvatar());
 	}
 
 	private function getRecirculationContentType( $contentType ) {
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/services/CakeRelatedContentService.php:170]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/fc569d281496e31233952a2ca3ad14cbcb555373/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/fc569d281496e31233952a2ca3ad14cbcb555373/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/fc569d281496e31233952a2ca3ad14cbcb555373/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
- MAJOR 2 major
- MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [CakeRelatedContentService.php#L28](https://github.com/Wikia/app/blob/d8c41c4b319239fdd26f773cf104ac9e9f37e623/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L28): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
2. MINOR [CakeRelatedContentService.php#L41](https://github.com/Wikia/app/blob/d8c41c4b319239fdd26f773cf104ac9e9f37e623/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L41): Define a constant instead of duplicating this literal "title" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [CakeRelatedContentService.php#L42](https://github.com/Wikia/app/blob/d8c41c4b319239fdd26f773cf104ac9e9f37e623/extensions/wikia/Recirculation/services/CakeRelatedContentService.php#L42): Define a constant instead of duplicating this literal "limit" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/Recirculation/services/CakeRelatedContentService.php (hunk 4 above)
Comment (tiffperumpail) [reply extensions/wikia/Recirculation/services/CakeRelatedContentService.php]: Fixed in fc569d281496e31233952a2ca3ad14cbcb555373

File: extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js (hunk 3 above)
Comment (tiffperumpail) [reply extensions/wikia/Recirculation/js/helpers/CakeRelatedContentHelper.js]: Fixed in 2f79991f0654b40277978274b077c2b5d2f8f983

**Pull request closed without merging**